python/paddle/framework/io.py

```diff
@@ -241,7 +302,43 @@ def save(obj, path, pickle_protocol=2):
             opt_state_dict = adam.state_dict()
             paddle.save(opt_state_dict, "adam.pdopt")
     '''
+    # 1. input check
+    filename = os.path.basename(path)
+    if filename == "":
+        raise ValueError("The input path MUST be format of dirname/filename "
+                         "[dirname\\filename in Windows system], but received "
+                         "filename is empty string.")
+
+    # 2. save object
+    dirname = os.path.dirname(path)
+    if dirname and not os.path.exists(dirname):
+        os.makedirs(dirname)
+
+    use_binary_format = configs.get('use_binary_format', False)
+    if not isinstance(use_binary_format, bool):
+        raise TypeError(
+            "Type of `use_binary_format` should be bool, but received {}.".
+            format(type(use_binary_format)))
+
+    # If the user uses both 'pickle_protocol' and 'protocol', 'pickle_protocol' will be used.
```

Comment:
same above, protocol need to be used, 'pickle_protocol' is a deprecated arg

Reply:
Done,thx.